workspace_client.go

```diff
@@ -791,6 +795,11 @@ func NewWorkspaceClient(c ...*Config) (*WorkspaceClient, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if cfg.AccountID == "" || cfg.IsAccountClient() {
```

Comment:
this will not work with Terraform, as sometimes we know host only after upstream modules are initialized.

Reply:
Similar kind of check exist in `account_client.go` -

```
if cfg.AccountID == "" || !cfg.IsAccountClient() {
	return nil, ErrNotAccountClient
}
```

Can you help me understand the optimal solution then? to make it work with Terraform as well.

Reply:
terraform resolves config upon the first request

Reply:
Please review the latest changes and help with proper feedback